ch3/exercise3-11/comma.go

```diff
@@ -16,37 +16,35 @@ func main() {
 
 func comma(s string) string {
 
-	pointMatch, err := regexp.MatchString("\\.", s)
-	if err != nil {
-		return s + ": " + err.Error()
-	}
-	signMatch, err := regexp.MatchString("[+-]", s)
-	if err != nil {
-		return s + ": " + err.Error()
-	}
 
+	pointRe := regexp.MustCompile("\\.")
+	signRe:= regexp.MustCompile("[+-]")
+
+	pointMatch := pointRe.FindString(s)
+	signMatch := signRe.FindString(s)
```

Comment:
compiled regex have the `Match` function if you want a similar boolean match object.  `FindString` has a different signature (as you noticed when all the conditionals needed to be updates.